src/libs/types-rs/src/mount.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2019-2021 Alibaba Cloud
+// Copyright (c) 2019-2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Prefix to mark a volume as Kata special.
+pub const KATA_VOLUME_TYPE_PREFIX: &str = "kata:";
+
+/// The Mount should be ignored by the host and handled by the guest.
+pub const KATA_GUEST_MOUNT_PREFIX: &str = "kata:guest-mount:";
+
+/// KATA_EPHEMERAL_DEV_TYPE creates a tmpfs backed volume for sharing files between containers.
+pub const KATA_EPHEMERAL_VOLUME_TYPE: &str = "kata:ephemeral";
+
+/// KATA_HOST_DIR_TYPE use for host empty dir
+pub const KATA_HOST_DIR_VOLUME_TYPE: &str = "kata:hostdir";
+
+/// Information about a mount.
+#[derive(Debug, Clone, Default, serde::Serialize, serde::Deserialize)]
+pub struct Mount {
+    /// The source path for the mountpoint.
+    pub source: String,
+    /// The destination path for the mountpoint.
+    pub destination: String,
+    /// The type of filesystem for the mountpoint.
+    pub fs_type: String,
+    /// Mount options for the mountpoint.
+    pub options: Vec<String>,
+    /// Device id for device associated with the mountpoint.
+    pub device_id: String,
```

Comment:
RunD has implemented an optimization for volumes. If the volume is a dedicated block device,  runD directly passes through the block device into guest vm and mount the fs within vm, instead of mount the block device on host and passthrough fs to guest by virtio-fs/9p.

`device_id` is used to store information about the block device.

Reply:
Ok.  As usual this would be clearer if you did the unoptimized version first, then had the optimization as a separate patch, with its own commit message and rationale.  Changing the comment so it actually says what's in the field more specifically would be good too: "device id" could mean basically anything.

Reply:
Ideally we should follow the way to do it incrementally. But we are facing some issues here:
1) the schedule is very tight to get it ready before June.
2) there are tens of thousands lines of code pending for review:(
3) If we split a submodule into several pieces, it's a little hard for us to track the status. And though we are rewriting the code, we also need to keep compatible as much as possible with our internal implementations.
So how about upstream code in unit of submodules?